src/lib.rs

```diff
@@ -353,104 +321,102 @@ impl Serialize for GUID {
 }
 
 impl fmt::Display for GUID {
-	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-		let mut s1 = String::new();
-		for i in 0..2 {
-			s1.push_str(&format!("{:02X}", self.data4()[i]))
-		}
-		let mut s2 = String::new();
-		for i in 2..8 {
-			s2.push_str(&format!("{:02X}", self.data4()[i]))
-		}
-		write!(
-			f,
-			"{:08X}-{:04X}-{:04X}-{}-{}",
-			self.data1(),
-			self.data2(),
-			self.data3(),
-			s1,
-			s2
-		)
-	}
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "{:08X}-{:04X}-{:04X}-{:02X}{:02X}-{:02X}{:02X}{:02X}{:02X}{:02X}{:02X}",
+            self.data1(),
+            self.data2(),
+            self.data3(),
+            self.data4()[0],
+            self.data4()[1],
+            self.data4()[2],
+            self.data4()[3],
+            self.data4()[4],
+            self.data4()[5],
+            self.data4()[6],
+            self.data4()[7]
```

Comment:
Maybe use [`from_be_bytes`](https://doc.rust-lang.org/std/primitive.u64.html#method.from_be_bytes)?

Reply:
How would you use a `u64`? Yes there are 8 bytes but they are plot in the string via a `-`, so __`u16-u32u16`__ Is what I see as possible

Reply:
Ah right, I missed that. Maybe your old implementation was better, until something like https://doc.rust-lang.org/std/primitive.array.html#method.split_array_ref goes stable.

Reply:
Not really, reducing the amount of parts needing to go through fmt also decreases the amount of cycles spent checking, styling, guessing its size and in the end concatenation. Moving from single bytes to `u16-u32u16 ` got about a 120ms reduction